pkg/cmd/cli/cmd/login.go

```diff
@@ -178,6 +183,43 @@ func (o LoginOptions) Validate(args []string, serverFlag string) error {
 	return nil
 }
 
+func (o LoginOptions) checkConfigIsWritable(f *osclientcmd.Factory) (bool, error) {
+	kubeconfig := o.PathOptions.GetDefaultFilename()
+
+	file, err := os.OpenFile(kubeconfig, os.O_CREATE|os.O_WRONLY, 0600)
+	if err == nil {
+		file.Close()
+		return true, nil
+	}
+
+	// Whole directory does not exist and we cannot create kubeconfig file.
+	if os.IsNotExist(err) {
+		return true, nil
+	}
+
+	errmsg := kubeconfig
+
+	if os.IsPermission(err) {
+		errmsg += ": The kubeconfig file is not writable. Specify a writeable path with --config or by setting $KUBECONFIG."
```

Comment:
@legionus Do you prefer this flow with `CheckErr` or the one with simply writing to stderr?  This impl looks buggy, so I'm guessing the latter.  You can have the option: this is an exceptional path, where we want a certain kind of error to be presented to the user in a command specific manner.  It is not the "normal" `CheckErr` flow where we got a generic message that we want to make user-understandable in a standard way.